test/dlc_tests/sanity/test_telemetry.py

```diff
@@ -301,7 +305,7 @@ def _run_tag_failure_IMDSv2_disabled_as_hop_limit_1(image_uri, ec2_client, ec2_i
     preexisting_ec2_instance_tags = ec2_utils.get_ec2_instance_tags(ec2_instance_id, ec2_client=ec2_client)
     LOGGER.info(f"preexisting_ec2_instance_tags: {preexisting_ec2_instance_tags}")
 
-    ec2_utils.enforce_IMDSv2(ec2_instance_id)
+    ec2_utils.enforce_IMDSv2(ec2_instance_id, hop_limit=1)
 
```

Comment:
Lets delete the `enforce_IMDSv2` function as well

Reply:
I see that it is needed to enforce IMDSv2 for specific cases when we enable IMDSv1 and want to enable IMDSv2 again

Reply:
@jeet4320 Wouldn't the telemetry call fallback to IMDSv1? It looks like this is causing the tag to be applied on the instance even when it shouldn't.

Reply:
yes you are right

Reply:
Actually even it falls back, if IMDSV2 is enforced with hop limit 1 then IMDSv1 wont work. So for IMDSv2 case with hop limit 1, there wont be telemetry call. Thats this case comes under failure test